### PR TITLE
feat(capabilities): add new collections route

### PR DESCRIPTION
**Pull request by matthv**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'json'
+module ForestAdminAgent
+  module Routes
+    module Capabilities
+      class Collections < AbstractRoute
```
Comment (jbarreau) [inline packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb:5]: Is this route is accessible without authentication ? Or it's just me reading ruby ?

File: packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    'post',
+                    '/_internal/capabilities',
+                    ->(args) { handle_request(args) })
+
+          self
+        end
+
+        def handle_request(args = {})
+          @datasource = ForestAdminAgent::Facades::Container.datasource
+          collections = args[:params]['collectionNames'] || @datasource.collections.keys
```
Comment (jbarreau) [inline packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb:19]: We said the frontend always sent `collectionNames`, I thought we said to **not** compute filters over all collections if `collectionNames` is missing. Am I wrong ?

File: packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb (hunk 1 above)
Comment (matthv) [reply packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb]: well observed. It's fixed

File: packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb (hunk 2 above)
Comment (matthv) [reply packages/forest_admin_agent/lib/forest_admin_agent/routes/capabilities/collections.rb]: fixed !

**Review by nicolasalexandre9 — APPROVED** [review]
LGTM 👍

**Pull request merged**